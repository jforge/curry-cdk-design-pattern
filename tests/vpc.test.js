"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const assert_1 = require("@aws-cdk/assert");
const cdk = require("@aws-cdk/core");
const Vpc = require("../lib/vpc");
require("@aws-cdk/assert/jest");
const iVpc = {
    maxAzs: 2
};
test('Fine-Grained Assertions', () => {
    const app = new cdk.App();
    // WHEN
    const stack = new Vpc.VpcStack(app, 'VpcStack', iVpc);
    // THEN
    expect(stack).toHaveResource('AWS::EC2::VPC', {
        EnableDnsHostnames: true,
        EnableDnsSupport: true,
        InstanceTenancy: "default",
    });
});
test('Snapshot Tests', () => {
    const app = new cdk.App();
    const stack = new Vpc.VpcStack(app, 'VpcStack', iVpc);
    expect(assert_1.SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});
test('Validation Tests - maxAzs = 0 is Error', () => {
    const app = new cdk.App();
    expect(() => {
        new Vpc.VpcStack(app, 'VpcStack', {
            maxAzs: 0
        });
    }).toThrowError('maxAzs must be at least 2.');
});
test('Validation Tests - maxAzs = 1 is Error', () => {
    const app = new cdk.App();
    expect(() => {
        new Vpc.VpcStack(app, 'VpcStack', {
            maxAzs: 1
        });
    }).toThrowError('maxAzs must be at least 2.');
});
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoidnBjLnRlc3QuanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyJ2cGMudGVzdC50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOztBQUFBLDRDQUE2QztBQUM3QyxxQ0FBc0M7QUFDdEMsa0NBQW1DO0FBQ25DLGdDQUE4QjtBQUU5QixNQUFNLElBQUksR0FBRztJQUNYLE1BQU0sRUFBRSxDQUFDO0NBQ1YsQ0FBQTtBQUVELElBQUksQ0FBQyx5QkFBeUIsRUFBRSxHQUFHLEVBQUU7SUFDakMsTUFBTSxHQUFHLEdBQUcsSUFBSSxHQUFHLENBQUMsR0FBRyxFQUFFLENBQUM7SUFDMUIsT0FBTztJQUNQLE1BQU0sS0FBSyxHQUFHLElBQUksR0FBRyxDQUFDLFFBQVEsQ0FBQyxHQUFHLEVBQUUsVUFBVSxFQUFFLElBQUksQ0FBQyxDQUFDO0lBQ3RELE9BQU87SUFDUCxNQUFNLENBQUMsS0FBSyxDQUFDLENBQUMsY0FBYyxDQUFDLGVBQWUsRUFBRTtRQUM1QyxrQkFBa0IsRUFBRSxJQUFJO1FBQ3hCLGdCQUFnQixFQUFFLElBQUk7UUFDdEIsZUFBZSxFQUFFLFNBQVM7S0FDM0IsQ0FBQyxDQUFDO0FBQ1AsQ0FBQyxDQUFDLENBQUM7QUFFSCxJQUFJLENBQUMsZ0JBQWdCLEVBQUUsR0FBRyxFQUFFO0lBQzFCLE1BQU0sR0FBRyxHQUFHLElBQUksR0FBRyxDQUFDLEdBQUcsRUFBRSxDQUFDO0lBQzFCLE1BQU0sS0FBSyxHQUFHLElBQUksR0FBRyxDQUFDLFFBQVEsQ0FBQyxHQUFHLEVBQUUsVUFBVSxFQUFFLElBQUksQ0FBQyxDQUFDO0lBQ3RELE1BQU0sQ0FBQyxtQkFBVSxDQUFDLGdCQUFnQixDQUFDLEtBQUssQ0FBQyxDQUFDLENBQUMsZUFBZSxFQUFFLENBQUM7QUFDL0QsQ0FBQyxDQUFDLENBQUM7QUFFSCxJQUFJLENBQUMsd0NBQXdDLEVBQUUsR0FBRyxFQUFFO0lBQ2xELE1BQU0sR0FBRyxHQUFHLElBQUksR0FBRyxDQUFDLEdBQUcsRUFBRSxDQUFDO0lBQzFCLE1BQU0sQ0FBQyxHQUFHLEVBQUU7UUFDVixJQUFJLEdBQUcsQ0FBQyxRQUFRLENBQUMsR0FBRyxFQUFFLFVBQVUsRUFBRTtZQUNoQyxNQUFNLEVBQUUsQ0FBQztTQUNWLENBQUMsQ0FBQztJQUNMLENBQUMsQ0FBQyxDQUFDLFlBQVksQ0FBQyw0QkFBNEIsQ0FBQyxDQUFDO0FBQ2hELENBQUMsQ0FBQyxDQUFDO0FBRUgsSUFBSSxDQUFDLHdDQUF3QyxFQUFFLEdBQUcsRUFBRTtJQUNsRCxNQUFNLEdBQUcsR0FBRyxJQUFJLEdBQUcsQ0FBQyxHQUFHLEVBQUUsQ0FBQztJQUMxQixNQUFNLENBQUMsR0FBRyxFQUFFO1FBQ1YsSUFBSSxHQUFHLENBQUMsUUFBUSxDQUFDLEdBQUcsRUFBRSxVQUFVLEVBQUU7WUFDaEMsTUFBTSxFQUFFLENBQUM7U0FDVixDQUFDLENBQUM7SUFDTCxDQUFDLENBQUMsQ0FBQyxZQUFZLENBQUMsNEJBQTRCLENBQUMsQ0FBQztBQUNoRCxDQUFDLENBQUMsQ0FBQyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCB7IFN5bnRoVXRpbHMgfSBmcm9tICdAYXdzLWNkay9hc3NlcnQnO1xuaW1wb3J0IGNkayA9IHJlcXVpcmUoJ0Bhd3MtY2RrL2NvcmUnKTtcbmltcG9ydCBWcGMgPSByZXF1aXJlKCcuLi9saWIvdnBjJyk7XG5pbXBvcnQgJ0Bhd3MtY2RrL2Fzc2VydC9qZXN0JztcblxuY29uc3QgaVZwYyA9IHtcbiAgbWF4QXpzOiAyXG59XG5cbnRlc3QoJ0ZpbmUtR3JhaW5lZCBBc3NlcnRpb25zJywgKCkgPT4ge1xuICAgIGNvbnN0IGFwcCA9IG5ldyBjZGsuQXBwKCk7XG4gICAgLy8gV0hFTlxuICAgIGNvbnN0IHN0YWNrID0gbmV3IFZwYy5WcGNTdGFjayhhcHAsICdWcGNTdGFjaycsIGlWcGMpO1xuICAgIC8vIFRIRU5cbiAgICBleHBlY3Qoc3RhY2spLnRvSGF2ZVJlc291cmNlKCdBV1M6OkVDMjo6VlBDJywge1xuICAgICAgRW5hYmxlRG5zSG9zdG5hbWVzOiB0cnVlLFxuICAgICAgRW5hYmxlRG5zU3VwcG9ydDogdHJ1ZSxcbiAgICAgIEluc3RhbmNlVGVuYW5jeTogXCJkZWZhdWx0XCIsXG4gICAgfSk7XG59KTtcblxudGVzdCgnU25hcHNob3QgVGVzdHMnLCAoKSA9PiB7XG4gIGNvbnN0IGFwcCA9IG5ldyBjZGsuQXBwKCk7XG4gIGNvbnN0IHN0YWNrID0gbmV3IFZwYy5WcGNTdGFjayhhcHAsICdWcGNTdGFjaycsIGlWcGMpO1xuICBleHBlY3QoU3ludGhVdGlscy50b0Nsb3VkRm9ybWF0aW9uKHN0YWNrKSkudG9NYXRjaFNuYXBzaG90KCk7XG59KTtcblxudGVzdCgnVmFsaWRhdGlvbiBUZXN0cyAtIG1heEF6cyA9IDAgaXMgRXJyb3InLCAoKSA9PiB7XG4gIGNvbnN0IGFwcCA9IG5ldyBjZGsuQXBwKCk7XG4gIGV4cGVjdCgoKSA9PiB7XG4gICAgbmV3IFZwYy5WcGNTdGFjayhhcHAsICdWcGNTdGFjaycsIHtcbiAgICAgIG1heEF6czogMFxuICAgIH0pO1xuICB9KS50b1Rocm93RXJyb3IoJ21heEF6cyBtdXN0IGJlIGF0IGxlYXN0IDIuJyk7XG59KTtcblxudGVzdCgnVmFsaWRhdGlvbiBUZXN0cyAtIG1heEF6cyA9IDEgaXMgRXJyb3InLCAoKSA9PiB7XG4gIGNvbnN0IGFwcCA9IG5ldyBjZGsuQXBwKCk7XG4gIGV4cGVjdCgoKSA9PiB7XG4gICAgbmV3IFZwYy5WcGNTdGFjayhhcHAsICdWcGNTdGFjaycsIHtcbiAgICAgIG1heEF6czogMVxuICAgIH0pO1xuICB9KS50b1Rocm93RXJyb3IoJ21heEF6cyBtdXN0IGJlIGF0IGxlYXN0IDIuJyk7XG59KTtcbiJdfQ==