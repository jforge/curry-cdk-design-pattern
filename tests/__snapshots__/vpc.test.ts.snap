// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Tests 1`] = `
Object {
  "Resources": Object {
    "VpcStandardB3B9526E": Object {
      "Properties": Object {
        "CidrBlock": "172.16.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/VpcStandard",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "VpcStandardIGWDB693EAA": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/VpcStandard",
          },
        ],
      },
      "Type": "AWS::EC2::InternetGateway",
    },
    "VpcStandardVPCGWB9B62BE0": Object {
      "Properties": Object {
        "InternetGatewayId": Object {
          "Ref": "VpcStandardIGWDB693EAA",
        },
        "VpcId": Object {
          "Ref": "VpcStandardB3B9526E",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
    "VpcStandardprivateSubnet1DefaultRoute55C8A667": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "VpcStandardpublicSubnet1NATGateway564B2D77",
        },
        "RouteTableId": Object {
          "Ref": "VpcStandardprivateSubnet1RouteTable84D78A4C",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VpcStandardprivateSubnet1RouteTable84D78A4C": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/VpcStandard/privateSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "VpcStandardB3B9526E",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VpcStandardprivateSubnet1RouteTableAssociation831FB396": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "VpcStandardprivateSubnet1RouteTable84D78A4C",
        },
        "SubnetId": Object {
          "Ref": "VpcStandardprivateSubnet1Subnet94A66483",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VpcStandardprivateSubnet1Subnet94A66483": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "172.16.2.0/24",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/VpcStandard/privateSubnet1",
          },
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
        ],
        "VpcId": Object {
          "Ref": "VpcStandardB3B9526E",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VpcStandardprivateSubnet2DefaultRoute379F4F4F": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "VpcStandardpublicSubnet2NATGateway6EC1566C",
        },
        "RouteTableId": Object {
          "Ref": "VpcStandardprivateSubnet2RouteTable143AFB99",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VpcStandardprivateSubnet2RouteTable143AFB99": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/VpcStandard/privateSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "VpcStandardB3B9526E",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VpcStandardprivateSubnet2RouteTableAssociationD13F88B1": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "VpcStandardprivateSubnet2RouteTable143AFB99",
        },
        "SubnetId": Object {
          "Ref": "VpcStandardprivateSubnet2Subnet6050B502",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VpcStandardprivateSubnet2Subnet6050B502": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            1,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "172.16.3.0/24",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/VpcStandard/privateSubnet2",
          },
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
        ],
        "VpcId": Object {
          "Ref": "VpcStandardB3B9526E",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VpcStandardpublicSubnet1DefaultRoute52E212C1": Object {
      "DependsOn": Array [
        "VpcStandardVPCGWB9B62BE0",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "VpcStandardIGWDB693EAA",
        },
        "RouteTableId": Object {
          "Ref": "VpcStandardpublicSubnet1RouteTable9F5AF121",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VpcStandardpublicSubnet1EIP2999EF60": Object {
      "Properties": Object {
        "Domain": "vpc",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/VpcStandard/publicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "VpcStandardpublicSubnet1NATGateway564B2D77": Object {
      "Properties": Object {
        "AllocationId": Object {
          "Fn::GetAtt": Array [
            "VpcStandardpublicSubnet1EIP2999EF60",
            "AllocationId",
          ],
        },
        "SubnetId": Object {
          "Ref": "VpcStandardpublicSubnet1Subnet65900542",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/VpcStandard/publicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "VpcStandardpublicSubnet1RouteTable9F5AF121": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/VpcStandard/publicSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "VpcStandardB3B9526E",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VpcStandardpublicSubnet1RouteTableAssociation7669FDDF": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "VpcStandardpublicSubnet1RouteTable9F5AF121",
        },
        "SubnetId": Object {
          "Ref": "VpcStandardpublicSubnet1Subnet65900542",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VpcStandardpublicSubnet1Subnet65900542": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "172.16.0.0/24",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/VpcStandard/publicSubnet1",
          },
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
        ],
        "VpcId": Object {
          "Ref": "VpcStandardB3B9526E",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VpcStandardpublicSubnet2DefaultRoute8B970526": Object {
      "DependsOn": Array [
        "VpcStandardVPCGWB9B62BE0",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "VpcStandardIGWDB693EAA",
        },
        "RouteTableId": Object {
          "Ref": "VpcStandardpublicSubnet2RouteTableD367FA3F",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VpcStandardpublicSubnet2EIP93F31E94": Object {
      "Properties": Object {
        "Domain": "vpc",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/VpcStandard/publicSubnet2",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "VpcStandardpublicSubnet2NATGateway6EC1566C": Object {
      "Properties": Object {
        "AllocationId": Object {
          "Fn::GetAtt": Array [
            "VpcStandardpublicSubnet2EIP93F31E94",
            "AllocationId",
          ],
        },
        "SubnetId": Object {
          "Ref": "VpcStandardpublicSubnet2Subnet42CBD72F",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/VpcStandard/publicSubnet2",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "VpcStandardpublicSubnet2RouteTableAssociation3662BF44": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "VpcStandardpublicSubnet2RouteTableD367FA3F",
        },
        "SubnetId": Object {
          "Ref": "VpcStandardpublicSubnet2Subnet42CBD72F",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VpcStandardpublicSubnet2RouteTableD367FA3F": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/VpcStandard/publicSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "VpcStandardB3B9526E",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VpcStandardpublicSubnet2Subnet42CBD72F": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            1,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "172.16.1.0/24",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/VpcStandard/publicSubnet2",
          },
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
        ],
        "VpcId": Object {
          "Ref": "VpcStandardB3B9526E",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
  },
}
`;
